src/js/components/Tag/index.d.ts

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import {
+  A11yTitleType,
+  AlignSelfType,
+  BackgroundType,
+  GridAreaType,
+  MarginType,
+  // Omit,
+  PolymorphicType,
+} from '../../utils';
+
+export interface TagProps {
+  a11yTitle?: A11yTitleType;
+  alignSelf?: AlignSelfType;
+  gridArea?: GridAreaType;
+  name?: string;
+  value: string | number;
+  onClick?: (...args: any[]) => any;
+  onRemove?: (...args: any[]) => any;
+  size?: 'small' | 'medium' | 'large';
```

Comment:
I don't think we are handling `size` correctly. I think it ends up in `...rest` which is correctly propagated to the StyledTagButton but then incorrectly to the container Box in the onRemove case. I'm actually inclined to remove `size` for this initial version. We can add it later.

Reply:
Definitely wasn't going to expose size yet so I think I just need to remove it from index.d.ts